elkm1_lib/keypads.py

```diff
@@ -30,13 +36,17 @@ def __init__(self, connection: Connection, notifier: Notifier) -> None:
         notifier.attach("IC", self._ic_handler)
         notifier.attach("KA", self._ka_handler)
         notifier.attach("KC", self._kc_handler)
+        notifier.attach("KF", self._kf_handler)
         notifier.attach("LW", self._lw_handler)
         notifier.attach("ST", self._st_handler)
 
     def sync(self) -> None:
         """Retrieve areas from ElkM1"""
         self._connection.send(ka_encode())
         self.get_descriptions(TextDescriptions.KEYPAD.value)
+        # Send KF for each of our keypads
+        for keypad in self.elements:
+            self._connection.send(kf_encode(keypad.index))
```

Comment:
I have tested the code now and it works. I now think I understand why you asked about looping over the keypads here. It does seem redundant.

Here is what I understand. On the keypads `last_function_key` is set to `None`. This sync is called sending `FORCE_KF_SYNC` to each keypad. The responses then set the `last_function_key` to `FORCE_KF_SYNC`.

Meanwhile over in the areas code it receives 16 KF replies and sets the chime mode for each of the 8 areas 16 times.

If that's right, here is what I suggest. In the keypads code, init `last_function_key` to `FORCE_KF_SYNC`. Remove the kf sync code from keypads. Add kf sync code to keypads; a single message will do.

I'm open for discussion, these were my observations from testing.

Reply:
That sounds good

Reply:
Issue moving the send of kf_encode to area, I need the keypad index of one keypads for that area.  I'm not seeing how I can get that from inside he Areas class?